Description: Add loong64 support
Author: Han Gao <gaohan@iscas.ac.cn>

Last-Update: 2023-02-23

--- python3.11-3.11.2.orig/configure
+++ python3.11-3.11.2/configure
@@ -6073,6 +6073,16 @@ cat > conftest.c <<EOF
         hppa-linux-gnu
 # elif defined(__ia64__)
         ia64-linux-gnu
+# elif defined(__loongarch64)
+#  if defined(__loongarch_double_float)
+        loongarch64-linux-gnu
+#  elif defined(__loongarch_single_float)
+        loongarch64-linux-gnuf32
+#  elif defined(__loongarch_soft_float)
+        loongarch64-linux-gnusf
+#  else
+#   error unknown platform triplet
+#  endif
 # elif defined(__m68k__) && !defined(__mcoldfire__)
         m68k-linux-gnu
 # elif defined(__mips_hard_float) && defined(__mips_isa_rev) && (__mips_isa_rev >=6) && defined(_MIPSEL)
--- python3.11-3.11.2.orig/configure.ac
+++ python3.11-3.11.2/configure.ac
@@ -968,6 +968,16 @@ cat > conftest.c <<EOF
         hppa-linux-gnu
 # elif defined(__ia64__)
         ia64-linux-gnu
+# elif defined(__loongarch64)
+#  if defined(__loongarch_double_float)
+        loongarch64-linux-gnu
+#  elif defined(__loongarch_single_float)
+        loongarch64-linux-gnuf32
+#  elif defined(__loongarch_soft_float)
+        loongarch64-linux-gnusf
+#  else
+#   error unknown platform triplet
+#  endif
 # elif defined(__m68k__) && !defined(__mcoldfire__)
         m68k-linux-gnu
 # elif defined(__mips_hard_float) && defined(__mips_isa_rev) && (__mips_isa_rev >=6) && defined(_MIPSEL)
